### PR TITLE
Add frosh 2023 sponsor site

### PR DESCRIPTION
**Pull request by EarthenSky**
@micahdbak
 
This is a ported version of your sponsor site to the frosh website.

the site ~~is~~ will be located at https://sfucsss.org/events/frosh/2023/sponsor

~~I need to make sure that CI doesn't die at home later~~ Added the extra file & fixed CI, it looks good!

### CODE REVIEW

File: csss-site/src/events/frosh/urls.py [hunk 1]
```diff
@@ -45,5 +45,9 @@
     url(r'^2022$', frosh2022.index, name="Frosh Week 2022 Entrance"),
     url(r'^2022/frosh$', frosh2022.frosh, name="Frosh Week 2022"),
     url(r'^2022/secret$', frosh2022.secret, name="Not A Secret"),
+    # url(r'^2023$', frosh2023.index, name="Frosh Week 2023 Entrance"),
+    # url(r'^2023/frosh$', frosh2023.frosh, name="Frosh Week 2023"),
+    url(r'^2023/secret$', frosh2023.secret, name="Not A Secret"),
```
Comment (modernNeo) [inline csss-site/src/events/frosh/urls.py:50]: you did not add w3c tests for these two new pages 😄

**Review by modernNeo — CHANGES_REQUESTED** [review]

File: csss-site/src/events/frosh/urls.py (hunk 1 above)
Comment (EarthenSky) [reply csss-site/src/events/frosh/urls.py:50]: I was not aware I had to make any test cases? The wiki doesn't mention it.

Comment (modernNeo) [reply csss-site/src/events/frosh/urls.py:50]: >  The wiki doesn't mention it. 

fuck, fair point 😅 look [here](https://github.com/CSSS/csss-site/blob/master/csss_site_w3c/events/frosh/tests.py) for reference

**Comment by EarthenSky** [comment]
added w3c testcases & `python3 manage.py test` succeeds (unsure if this invocation is part of CI)

File: csss_site_w3c/events/frosh/tests.py [hunk 2]
```diff
@@ -169,5 +169,14 @@ def test_frosh_2022_secret_page(self):
         W3CValidation().validate_page(path="/events/frosh/2022/secret")
 
 
+    def test_frosh_2023_sponsor_page(self):
+        W3CValidation().validate_page(path="/events/frosh/2023/sponsor")
+
+    # the secret page is intended to not be valid html as an easter egg to new froshees
+    @skip("not ready for w3c validation")
```
Comment (modernNeo) [inline csss_site_w3c/events/frosh/tests.py:176]: uh, guess I forgot to mention but no new webpages are allowed to use the `@skip` annotation. I use that annotation just on existing pages that I have not had time to fix.

**Review by modernNeo — CHANGES_REQUESTED** [review]

File: csss_site_w3c/events/frosh/tests.py (hunk 2 above)
Comment (EarthenSky) [reply csss_site_w3c/events/frosh/tests.py:176]: I understand, however by design this webpage is supposed to return an error 70% of the time. It's an easter egg.

Comment (modernNeo) [reply csss_site_w3c/events/frosh/tests.py:176]: oh, then put that comment in the skip annotation itself

File: csss-site/src/events/frosh/urls.py (hunk 1 above)
Comment (modernNeo) [reply csss-site/src/events/frosh/urls.py:50]: no, you do `python -u manage.py test --testrunner xmlrunner.extra.djangotestrunner.XMLTestRunner`

Comment (EarthenSky) [reply csss-site/src/events/frosh/urls.py:50]: It's still not working, even after I installed the requirements.txt for `csss_site_w3c`

I don't know what values to set the following env vars:
- UNIT_TEST_URL
- W3C_TEST_RESULT_DIRECTORY

Comment (modernNeo) [reply csss-site/src/events/frosh/urls.py:50]: 😅 i'm sorry man, I think I need to make an issue to address this cause I completely forgot that folks might want to do this.

But in the meantime what I suggest is going to https://validator.w3.org/ and plug in the non-working pages into there
like https://validator.w3.org/nu/?doc=https%3A%2F%2Fdev.sfucsss.org%2FPR-592%2Fevents%2Ffrosh%2F2023%2Fsponsor

Comment (EarthenSky) [reply csss-site/src/events/frosh/urls.py:50]: sure thing, and thanks for the workaround!

**Comment by EarthenSky** [comment]
all w3c issues should be fixed, but the CI still has an error. Looking at the [Test Result](https://jenkins.sfucsss.org/job/website-validate-and-deploy/job/PR-592/8/testReport/) has no failures. I don't understand why CI failed.

**Comment by modernNeo** [comment]
> all w3c issues should be fixed, but the CI still has an error. Looking at the [Test Result](https://jenkins.sfucsss.org/job/website-validate-and-deploy/job/PR-592/8/testReport/) has no failures. I don't understand why CI failed.

That page you linked also indicates that no tests were run. which is because your build [#8] failed [to deploy to the test server so it couldn't even run the tests on it](https://jenkins.sfucsss.org/job/website-validate-and-deploy/job/PR-592/)

And that is because you just didn't update your branch after my latest CI changes to master.

**Comment by EarthenSky** [comment]
> That page you linked also indicates that no tests were run. 

huh, it says it passed a bunch of tests, and it said they all elapsed some amount of time (22s in total). I don't see it say that it didn't run any of the tests.
tests

Ah! I see here https://jenkins.sfucsss.org/job/website-validate-and-deploy/job/PR-592/ now it's a bit more clear. It did pass all the python tests, but failed to deploy afterwards.

**Comment by modernNeo** [comment]
just merged the fix into your PR so we'll see the latest results soon

**Comment by modernNeo** [comment]
@EarthenSky still failed 😞

**Comment by EarthenSky** [comment]
whew, luckily only two more warnings, and these are ones I can fix. I'll clean it up tomorrow when I get to campus. Thanks for the all the help!

**Comment by EarthenSky** [comment]
the errors should all be fixed now, but it's timing out when deploying to the CI server

**Comment by modernNeo** [comment]
> Hey, this is overly complicated for a static site. We need a better way to do this.

if you think this is overly complicated, you should see some of the setups I have come across in my FT jobs 😂

**Comment by EarthenSky** [comment]
> if you think this is overly complicated, you should see some of the setups I have come across in my FT jobs joy

Other setups being bad isn't reason for this to be bad as well. Honestly, w3c validation and our entire 5 minute CI run is overkill for a static html page. I would be willing to do it if CI didn't break, but at this point it's a bit much.... Do you mind if I disable the w3c tests & push to main (for now), until we get things sorted out?

**Comment by modernNeo** [comment]
jeez you are quick to give up. It's an OOM issue. I deleted the JSON that is throwing the error and re-triggered the build. apparently the website has so much data now, I will have to make it more clever about how it loads PROD data.

**Comment by modernNeo** [comment]
in essence, we should be celebrating how popular this blob of code is 😉

**Comment by modernNeo** [comment]
> Honestly, w3c validation and our entire 5 minute CI run is overkill for a static html page. 

if you have time to make the CI smarter and modular, feel free 😄

File: csss-site/src/csss/settings.py [hunk 3]
```diff
@@ -94,7 +94,7 @@
     HOST_ADDRESS = os.environ['HOST_ADDRESS']
 else:
     HOST_ADDRESS = "127.0.0.1"
-ALLOWED_HOSTS = [HOST_ADDRESS]
+ALLOWED_HOSTS = [HOST_ADDRESS, 'localhost']
```
Comment (modernNeo) [inline csss-site/src/csss/settings.py:97]: please undo this change.

File: csss-site/src/events/frosh/templates/frosh/2022/.frosh.html.swp
Comment (modernNeo) [inline csss-site/src/events/frosh/templates/frosh/2022/.frosh.html.swp:1]: remove this file

File: csss-site/src/events/frosh/templates/frosh/2023/sponsor/.index.html.swp
Comment (modernNeo) [inline csss-site/src/events/frosh/templates/frosh/2023/sponsor/.index.html.swp:1]: remove this file

File: csss-site/src/events/frosh/urls.py [hunk 4]
```diff
@@ -45,5 +45,9 @@
...
+    url(r'^2023/sponsor$', frosh2023.sponsor, name="For Sponsors"),
```
Comment (modernNeo) [inline csss-site/src/events/frosh/urls.py:51]: shouldn't this end with `sponsors`? it's not like we'll have only 1 of them right? and the function should also be `sponsors`

File: csss-site/src/events/frosh/views/frosh2023.py [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+def frosh(request):
+    return render(request, 'frosh/2023/frosh.html', create_frosh_context())
+
+
+def secret(request):
+    test = randrange(0, 10)
+    if test < 7:
+        raise Http404('Not sure if there is a secret here. Please refresh your page and try again.')
+    return redirect('https://www.youtube.com/watch?v=dQw4w9WgXcQ', permanent=True)
```
Comment (modernNeo) [inline csss-site/src/events/frosh/views/frosh2023.py:20]: i don't see the point to using `redirect` here over `HttpResponseRedirect`? and if i am not mistaken, using `redirect` [might actually be more unnecessary work in this case](https://stackoverflow.com/a/13305294)?

File: csss_site_w3c/events/frosh/tests.py [hunk 6]
```diff
@@ -169,5 +169,13 @@ def test_frosh_2022_secret_page(self):
...
+
+    @skip("This webpage is supposed to return an error 70% of the time, as an easter egg to froshees")
+    def test_frosh_2023_secret_page(self):
+        W3CValidation().validate_page(path="/events/frosh/2023/secret")
+
+
     def test_frosh_main_page(self):
-        W3CValidation().validate_page(path="/events/frosh")
\ No newline at end of file
+        W3CValidation().validate_page(path="/events/frosh")
```
Comment (modernNeo) [inline csss_site_w3c/events/frosh/tests.py:181]: undo this change please

**Review by modernNeo — CHANGES_REQUESTED** [review]

File: csss-site/src/events/frosh/urls.py (hunk 4 above)
Comment (modernNeo) [reply csss-site/src/events/frosh/urls.py:51]: in addition to this, in alot of places in your code, you use `sponsor` [like in folder names] when it should really be `sponsors` imo

File: csss-site/src/events/frosh/views/frosh2023.py (hunk 5 above)
Comment (EarthenSky) [reply csss-site/src/events/frosh/views/frosh2023.py:20]: I copied this from last year. All I know is that the code works

File: csss_site_w3c/events/frosh/tests.py (hunk 6 above)
Comment (EarthenSky) [reply csss_site_w3c/events/frosh/tests.py:181]: the newline?

File: csss-site/src/events/frosh/urls.py (hunk 4 above)
Comment (EarthenSky) [reply csss-site/src/events/frosh/urls.py:51]: this is the "sponsor site". I see what you mean by `sponsors` making sense, but the distinction is negligible

**Comment by EarthenSky** [comment]
> if you have time to make the CI smarter and modular, feel free 😄

I've been talking with some others about splitting off & simplifying the static side of the website. I agree that having a large testsuite and everything is positive for the elections system since it actually has server side behaviour! But it's definitely overkill for these subdomain static website purposes that we run for all the events.

Github pages for example is dead simple, and never fails. This is kind of system we want for these kinds of sites (and honestly, I'm not against the front page of our site being hosted from github pages, with the more complicated systems hosted elsewhere).

File: csss-site/src/csss/settings.py (hunk 3 above)
Comment (EarthenSky) [reply csss-site/src/csss/settings.py]: nice catch, thanks!

**Comment by modernNeo** [comment]
> But it's definitely overkill for these subdomain static website purposes that we run for all the events.

I think that's a miconception that yall may have just cause the automated w3c validation is a new addition with some bugs in it that I am working out.

But if there's something else that is an issue, I am happy to hear the specifics.

I am opposed to github pages cause it leads to 30 different repos all owned by different folks who are former people and which we have no control over.

File: csss-site/src/events/frosh/urls.py (hunk 4 above)
Comment (modernNeo) [reply csss-site/src/events/frosh/urls.py:51]: this is the site sponsors go to to see what the event is or the site that showcases our sponsors?

**Comment by EarthenSky** [comment]
> But if there's something else that is an issue, I am happy to hear the specifics.

I think we'll probably look into it and bring it up later but good to know!

> I am opposed to github pages cause it leads to 30 different repos all owned by different folks who are former people and which we have no control over.

We don't necessarily have to use github pages for static hosting, but it's a good example (& is convenient if it does work). I will add however that github pages applies to github organizations as well, not just users. We would have a single github pages site for the csss github organization. I totally agree having 30 different repos would be super painful, and is something we never want.

File: csss-site/src/events/frosh/views/frosh2023.py (hunk 5 above)
Comment (modernNeo) [reply csss-site/src/events/frosh/views/frosh2023.py:20]: > Also, make sure not to prematurely optimize

my man, you argued that this can be done via a github page, I doubt you're gonna be utilizing the additional options of `redirect` if that's the case 😂

Comment (modernNeo) [reply csss-site/src/events/frosh/views/frosh2023.py:20]: > I directly copied this from last year

things can slip by me, I ain't perfect 🙃

File: csss-site/src/events/frosh/urls.py (hunk 4 above)
Comment (EarthenSky) [reply csss-site/src/events/frosh/urls.py:51]: this is the site that faces external companies an encourages them to sponsor us.

File: csss_site_w3c/events/frosh/tests.py (hunk 6 above)
Comment (modernNeo) [reply csss_site_w3c/events/frosh/tests.py]: you fixed it, whatever the issue was

File: csss-site/src/events/frosh/urls.py (hunk 4 above)
Comment (modernNeo) [reply csss-site/src/events/frosh/urls.py:51]: ah, gotcha

File: csss-site/src/events/frosh/static/frosh_static/2023/sponsor/sponsors/bitquill.png
Comment (modernNeo) [inline csss-site/src/events/frosh/static/frosh_static/2023/sponsor/sponsors/bitquill.png:1]: are these photos of past sponsors?

File: CI/validate_and_deploy/Jenkinsfile
Comment (modernNeo) [inline CI/validate_and_deploy/Jenkinsfile:1]: think you did a rebase or pull wrong, do you need my help to fix?

File: csss-site/src/events/frosh/templates/frosh/2022/.frosh.html.swp
Comment (modernNeo) [inline csss-site/src/events/frosh/templates/frosh/2022/.frosh.html.swp:1]: how is this back in the PR?

File: csss-site/src/events/frosh/templates/frosh/2023/sponsor/.index.html.swp
Comment (modernNeo) [inline csss-site/src/events/frosh/templates/frosh/2023/sponsor/.index.html.swp:1]: same for this.

File: csss-site/src/events/frosh/views/frosh2023.py (hunk 5 above)
Comment (modernNeo) [reply csss-site/src/events/frosh/views/frosh2023.py:20]: please switch to `HttpResponseRedirect`. If you don't know how to. I can do it for you.

**Review by modernNeo — CHANGES_REQUESTED** [review]

File: csss-site/src/events/frosh/views/frosh2023.py (hunk 5 above)
Comment (EarthenSky) [reply csss-site/src/events/frosh/views/frosh2023.py:20]: I would prefer to leave it as `redirect` because it is clearer to read. Since this is likely only going to be run a few times ever, readability is more important than performance, in this case. As the stackoverflow you linked claimed, `redirect` is a more flexible alternative to `HttpResponseRedirect`.

File: CI/validate_and_deploy/Jenkinsfile
Comment (EarthenSky) [reply CI/validate_and_deploy/Jenkinsfile:1]: I didn't make any changes to the Jenkins file. My understanding is that you bumped it to 30s in af66ed1503a4e18903f2662d3636d1883bc9e8e3

File: csss-site/src/events/frosh/templates/frosh/2022/.frosh.html.swp
Comment (EarthenSky) [reply csss-site/src/events/frosh/templates/frosh/2022/.frosh.html.swp:1]: oh what the heck? No idea how it got back in. I must've missed it. I'll remove it now.

Comment (EarthenSky) [reply csss-site/src/events/frosh/templates/frosh/2022/.frosh.html.swp:1]: Ah yep, it looks like I missed it & never removed them. Nice catch, thanks!

File: csss-site/src/events/frosh/static/frosh_static/2023/sponsor/sponsors/bitquill.png
Comment (EarthenSky) [reply csss-site/src/events/frosh/static/frosh_static/2023/sponsor/sponsors/bitquill.png:1]: yes

File: csss-site/src/events/frosh/views/frosh2023.py (hunk 5 above)
Comment (modernNeo) [reply csss-site/src/events/frosh/views/frosh2023.py:20]: ordinarly, I would not care, but in this case, a precedent has already been set and the SO is actually a mark against you imo cause you're using something with more features than you need.

File: CI/validate_and_deploy/Jenkinsfile
Comment (modernNeo) [reply CI/validate_and_deploy/Jenkinsfile:1]: i know, that's why I said I think you rebased or pulled wrong since they're showing in this PR

File: csss-site/src/events/frosh/static/frosh_static/2023/sponsor/sponsors/bitquill.png
Comment (modernNeo) [reply csss-site/src/events/frosh/static/frosh_static/2023/sponsor/sponsors/bitquill.png:1]: then please rename the folder path from `2023/sponsor/sponsors/` to `2023/sponsor/past_sponsors/`

File: csss-site/src/events/frosh/views/frosh2023.py (hunk 5 above)
Comment (EarthenSky) [reply csss-site/src/events/frosh/views/frosh2023.py:20]: > I also like how redirect is shorter. So I'd use redirect over HttpResponseRedirect.
> 
> Both are fine to use though.

This is a quote from the SO. The only precedent I see being set is that one _should_ use `redirect` instead of `HttpResponseRedirect`, which is what's being done right here.

File: csss-site/src/events/frosh/static/frosh_static/2023/sponsor/sponsors/bitquill.png
Comment (EarthenSky) [reply csss-site/src/events/frosh/static/frosh_static/2023/sponsor/sponsors/bitquill.png:1]: no

File: csss-site/src/events/frosh/urls.py [hunk 7]
```diff
@@ -45,5 +45,9 @@
     url(r'^2022$', frosh2022.index, name="Frosh Week 2022 Entrance"),
     url(r'^2022/frosh$', frosh2022.frosh, name="Frosh Week 2022"),
     url(r'^2022/secret$', frosh2022.secret, name="Not A Secret"),
+    # url(r'^2023$', frosh2023.index, name="Frosh Week 2023 Entrance"),
```
Comment (modernNeo) [inline csss-site/src/events/frosh/urls.py:48]: i don't like this approach you have taken, i'd rather this be un-commented but instead redirect to a temporary page we already use for such occasions. any issues with me making the change myself?

Comment (EarthenSky) [reply csss-site/src/events/frosh/urls.py:48]: sure, go for it!

File: CI/validate_and_deploy/Jenkinsfile
Comment (EarthenSky) [reply CI/validate_and_deploy/Jenkinsfile:1]: ah, I assumed it was just a commit you made. I don't see any issues here, but if you'd like to fix it, feel free.

**Comment by EarthenSky** [comment]
The build is failing to deploy to the test-server for a mysterious reason again. Do you have any ideas what caused it?

I'm unsure what caused it to fail, as the console output for the deploy isn't very clear. The only change I've made since then has been to client side code, so perhaps it's due to the size of the source increasing again?

**Comment by modernNeo** [comment]
the staging server needs to be restarted quite often now-a-days, I suspect its because you have 3 PRs at the moment and I never thought we'd have more than 2 at a time 🤷🏿‍♀️

**Comment by EarthenSky** [comment]
> I never thought we'd have more than 2 at a time

ah heck, okay I'll avoid leaving PRs for too long. Though I suppose this is important to consider if we get more people working on the website

**Comment by EarthenSky** [comment]
Also, I checked the dev server, the style is good now, feel free to merge it!

**Comment by modernNeo** [comment]
>  Though I suppose this is important to consider if we get more people working on the website

God I hope so

**Review by modernNeo — APPROVED** [review]
